corehq/apps/locations/forms.py

```diff
@@ -243,28 +242,35 @@ def clean_site_code(self):
 
         return site_code
 
+    def _get_allowed_types(self, parent):
+        parent_type = parent.location_type if parent else None
+        return list(LocationType.objects
+                    .filter(domain=self.domain,
+                            parent_type=parent_type)
+                    .all())
+
     def clean_location_type(self):
         loc_type = self.cleaned_data['location_type']
-
-        try:
-            loc_type_obj = LocationType.objects.get(Q(code=loc_type)|Q(name=loc_type))
-            self.cleaned_data['location_type_object'] = loc_type_obj
-        except LocationType.DoesNotExist:
-            assert False, "LocationType '{}' not found".format(loc_type)
-
-        child_types = allowed_child_types(self.domain,
-                                          self.cleaned_data.get('parent'))
-        if not loc_type:
-            if len(child_types) == 1:
-                return child_types[0]
-            assert False, 'You must select a location type'
-
-        if not child_types:
+        allowed_types = self._get_allowed_types(self.cleaned_data.get('parent'))
+        if not allowed_types:
             raise forms.ValidationError(_('The selected parent location cannot have child locations!'))
-        elif loc_type not in child_types:
-            raise forms.ValidationError(_('Location type not valid for the selected parent.'))
 
-        return loc_type
+        if not loc_type:
+            if len(allowed_types) == 1:
+                loc_type_obj = allowed_types[0]
+            else:
+                assert False, 'You must select a location type'
```

Comment:
There are actually a bunch of untranslated strings on this class, just updated them.